lib/galaxy/web/framework/helpers/grids.py

```diff
@@ -160,13 +162,13 @@ def loads_recurse(item):
                     if isinstance( column_filter, list ):
                         # Filter is a list; process each item.
                         for filter in column_filter:
-                            if not isinstance( filter, basestring ):
-                                filter = unicode( filter ).encode("utf-8")
+                            if not isinstance( filter, string_types ):
+                                text_type( filter ).encode("utf-8")
```

Comment:
@nsoranzo 
ahh, of course.
but ... what is this code actually doing?
I don't see filter being re-used anywhere ?!
Should this be:
```
column_filter_encoded = [ ]
for filter in column_filter:
    if not isinstance( filter, string_types ):
        filter = text_type( filter ).encode("utf-8")
    column_filter_encoded.append( filter )
extra_url_args[ "f-" + column.key ] = dumps( column_filter_encoded )
```
?